src/hooks/useFetch.test.ts

```diff
@@ -0,0 +1,79 @@
+import { renderHook, waitFor, act } from "@testing-library/react";
+import { useFetch } from "./useFetch";
+
+describe("useFetch()", () => {
```

Comment:
It turned out that testing-library has its function to safely run hooks without UI components via React. It's pretty surprising how powerful testing-library is! With this function, you can basically mock the UI up to run the hooks themselves and you can access it from `result` body from the function. 

article: https://kentcdodds.com/blog/how-to-test-custom-react-hooks